schema/proto3/neighbor.proto

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+syntax = "proto3";
+
+package alcor.schema;
+
+option java_package = "com.futurewei.alcor.schema";
+option java_outer_classname = "Neighbor";
+
+import "common.proto";
+
+enum NeighborType { 
+    L2 = 0; // the default type
+    L3 = 1;
+    HOST_DVR_MAC = 2;
+}
+
+message NeighborConfiguration {
+    uint32 format_version = 1;
+    uint32 revision_number = 2;
+
+    string id = 3;
+    NetworkType network_type = 4;  // we may move this to subnet level
+    string project_id = 5;
+    string vpc_id = 6;
+    string name = 7;
+    string mac_address = 8;
+    string host_ip_address = 9;
+    NeighborType neighbor_type = 10; // L2 (default) or L3 or HOST_DVR_MAC
+
+    message HostDvrMacAddress {
+        string subnet_id = 1;
+        string host_dvr_mac_address = 2;
+    }
+
+    message FixedIp {
+        string subnet_id = 1;
+        string ip_address = 2;
+    }
+
+    message AllowAddressPair {
+        string ip_address = 1;
+        string mac_address = 2;
+    }
+
+    repeated HostDvrMacAddress host_dvr_mac_addresses = 11;
```

Comment:
Why do we have multiple host_dvr_mac_addresses under neighbor?

Reply:
for neighbor type (HOST_DVR), it is very possible that multiple new DVR comes up from different compute host, when a new router is created. This "repeated" field is an optimization instead of sending down mutiple neighbor config.